capa/features/extractors/elf.py

```diff
@@ -854,28 +854,60 @@ def detect_elf_os(f) -> str:
     """
     f: type Union[BinaryIO, IDAIO]
     """
-    elf = ELF(f)
-
-    osabi_guess = guess_os_from_osabi(elf)
-    logger.debug("guess: osabi: %s", osabi_guess)
-
-    ph_notes_guess = guess_os_from_ph_notes(elf)
-    logger.debug("guess: ph notes: %s", ph_notes_guess)
-
-    sh_notes_guess = guess_os_from_sh_notes(elf)
-    logger.debug("guess: sh notes: %s", sh_notes_guess)
-
-    linker_guess = guess_os_from_linker(elf)
-    logger.debug("guess: linker: %s", linker_guess)
-
-    abi_versions_needed_guess = guess_os_from_abi_versions_needed(elf)
-    logger.debug("guess: ABI versions needed: %s", abi_versions_needed_guess)
-
-    needed_dependencies_guess = guess_os_from_needed_dependencies(elf)
-    logger.debug("guess: needed dependencies: %s", needed_dependencies_guess)
-
-    symtab_guess = guess_os_from_symtab(elf)
-    logger.debug("guess: pertinent symbol name: %s", symtab_guess)
+    try:
+        elf = ELF(f)
+    except Exception as e:
+        logger.debug(f"Error parsing ELF file: {str(e)}")
```

Comment:
please avoid pre-formatting strings that are passed to a logger, because the logger has some optimizations.
```suggestion
        logger.debug(f"Error parsing ELF file: %s", e)
```

Reply:
also recommend this be emitted at the `warning` level, since its exceptional but doesn't cause the program to crash